docs/docsite/rst/contributor_guide.rst

```diff
@@ -159,3 +159,141 @@ Customization
+
+In addition, all module utils within the collection must contain a comment explaining this, like so:
+
+.. code-block:: python
+
+    # FOR INTERNAL COLLECTION USE ONLY
+    # The interfaces in this file are meant for use within the community.hashi_vault collection
+    # and may not remain stable to outside uses. Changes may be made in ANY release, even a bugfix release.
+    # See also: https://github.com/ansible/community/issues/539#issuecomment-780839686
+    # Please open an issue if you have questions about this.
+
+It is best to look at `existing auth methods <https://github.com/ansible-collections/community.hashi_vault/tree/main/plugins/module_utils>`_ to get a feel for how they are implemented.
+
+Class anatomy
+-------------
+
+Each auth method class should be named ``HashiVaultAuthMethod<MethodName>`` and inherit from ``HashiVaultAuthMethodBase``.
+
+The base class provides some common functionality, like standardizing a way to emit warnings and providing a common function for validating required options.
+
+An auth method must run the base class's ``__init__`` function.
+
+It must implement two methods:
+
+* ``validate()`` -- this method does everything it can to ensure that the requirements are met for performing authentication with this particular auth method. This may include checking for required options, validating the values of those options, pulling in additional information and context from the environment, preparing that information for use by ``authenticate()``, etc. Generally speaking, it should not contact Vault, and should minimize reliance on external sources and services, but that is a guideline and the details will depend on the specifics of the auth method in question. ``validate()`` raises an exception if validation fails. If it succeeds, nothing is returned.
+* ``authenticate(client, use_token=False)`` -- this method performs the actual authentication, and it returns the API result of the authentication (which will include the token, lease information, etc.). The HVAC client object is passed in, as well an optional parameter ``use_token`` which specifies whether the client should have its token field set to the result of authentication (typically this is desired).
+
+The auth method class should also contain two fields:
+
+* ``NAME`` -- the name of the auth method.
+* ``OPTIONS`` -- a list containing the name of every option that may be used by the auth method, including optional options; this list should not include the ``auth_method`` option.
+
+Raising exceptions and warnings
+-------------------------------
+
+Becuase auth methods are shared among both Ansible modules and Ansible plugins, any exceptions raised must be applicable to both. Standard Python exceptions like ``KeyError`` can be raised if they appropriate.
```

Comment:
```suggestion
Because auth methods are shared among both Ansible modules and Ansible plugins, any exceptions raised must be applicable to both. Standard Python exceptions like ``KeyError`` can be raised if they appropriate.
```